lex/lexer_test.go

```diff
@@ -56,7 +56,7 @@ func TestLexer(t *testing.T) {
 		l.Push("fake", LexNumber)
 	}
 	// Should not infinitely push
-	assert.Equal(t, 500, len(l.stack))
+	assert.Equal(t, 600, len(l.stack))
```

Comment:
This is testing that, when we hit the max, we stop. I'd move 1000 to be a package-level private `const`, set the value above to be 1100 instead of 600, and ensure this equal that 1000 const.